documentation/Calls.md

```diff
@@ -0,0 +1,636 @@
+by the standard in cases where pointers to the original target
+data are required to be valid across the call (15.5.2.4(9-10)).
+In particular, compaction of assumed-shape arrays for discretionary
+contiguity on the leading dimension to ease SIMD vectorization
+cannot be done safely for `TARGET` dummies without `VALUE`.
+
+Effective arguments associated with known `INTENT(OUT)` dummies that
+require allocation of a temporary -- and this can only be for reasons of
+contiguity -- don't have to populate it, but they do have to perform
+minimal initialization of any `ALLOCATABLE` components so that
+the runtime doesn't crash when the callee finalizes and deallocates
+them.
+Note that calls to implicit interfaces must conservatively allow
+for the use of `INTENT(OUT)` by the callee.
+
+Except for `VALUE` and known `INTENT(IN)` dummy arguments, the original
+contents of local designators that have been compacted into temporaries
+could optionally have their `ALLOCATABLE` components invalidated
+across the call as an aid to debugging.
+
+Except for `VALUE` and known `INTENT(IN)` dummy arguments, the contents of
+the temporary storage will be copied back into the effective argument
+designator after control returns from the procedure, and it may be necessary
+to preserve addresses (or the values of subscripts and cosubscripts
+needed to recalculate them) of the effective argument designator, or its
+elements, in additional temporary storage if they can't be safely or
+quickly recomputed after the call.
+
+### `INTENT(OUT)` preparation
+
+Effective arguments that are associated with `INTENT(OUT)`
+dummy arguments are required to be definable.
+This cannot always be checked, as the use of `INTENT(OUT)`
+does not by itself mandate the use of an explicit interface.
+
+`INTENT(OUT)` arguments are finalized (as if) on entry to the called
+procedure.  In particular, in calls to elemental procedures,
+the elements of an array are finalized by a scalar or elemental
+`FINAL` procedure (7.5.6.3(7)).
+
+Derived type components that are `ALLOCATABLE` are finalized
+and deallocated.
+Components with initializers are (re)initialized.
+
+The preparation of effective arguments for `INTENT(OUT)` could be
+done on either side of the call.  If the preparation is
+done by the caller, there is an optimization opportunity
+in situations where unmodified incoming `INTENT(OUT)` dummy
+arguments are being passed onward as outgoing `INTENT(OUT)`
+arguments.
+
+### Arguments and function results requiring descriptors
+
+Dummy arguments are represented with the addresses of descriptors
+when they have any of the following characteristics:
+
+1. assumed-shape array (`DIMENSION(:)`)
+1. assumed-rank array (`DIMENSION(..)`)
+1. parameterized derived type with assumed `LEN` parameters
+1. polymorphic (`CLASS(T)`, `CLASS(*)`)
+1. assumed-type (`TYPE(*)`)
+1. coarray dummy argument
+1. `INTENT(IN) POINTER` argument (15.5.2.7, C.10.4)
```

Comment:
Doesn't all `POINTER` or `ALLOCATABLE` dummy arguments also need to be represented as descriptors ?
Or is this list just listing cases where the dummy needs a descriptor when the actual argument may not already have had one ?